src/core/error.js

```diff
@@ -81,3 +81,20 @@ export function rethrowAsync(var_args) {
     throw error;
   });
 }
+
+/**
+ * Executes the provided callback in a try/catch and rethrows any errors
+ * asynchronously.
+ *
+ * @param {function(...*):T} callback
+ * @param {...*} var_args
+ * @return {T}
+ * @template T
+ */
+export function tryCallback(callback, var_args) {
+  try {
+    return callback.apply(null, Array.prototype.slice.call(arguments, 1));
+  } catch (e) {
+    rethrowAsync(e);
```

Comment:
Do all callers of `tryCallback` want an out-of-band error?

Reply:
Yes, this is a semi-common pattern to `rethrowAsync` in the catch block